install.sh

```diff
@@ -131,44 +132,109 @@ check_deprecation() {
+        git config --global githooks.previousSearchDir "$OLD_CONFIG_VALUE" || LEGACY_TRANSFORM_FAILURES="true"
     fi
 
     # Copy legacy file to new location
     if [ -f "$INSTALL_DIR/autoupdate/registered" ]; then
-        cp "$INSTALL_DIR/autoupdate/registered" "$INSTALL_DIR/registered"
+        cp "$INSTALL_DIR/autoupdate/registered" "$INSTALL_DIR/registered" || LEGACY_TRANSFORM_FAILURES="true"
+    fi
+
+    # Check if we need to move .shared files entries to `--local githooks.shared`
+    # and split comma-separated `--global githooks.shared`
+
+    if [ "$LEGACY_TRANSFORM_FAILURES" = "true" ]; then
+        echo "! There were legacy transform errors: check stderr"
+        return 1
     fi
 
     return 0
 }
 
+############################################################
+# Tests if the commit sha `$1`
+#   is before or equal of the commit sha `$2`.
+#
+# Returns: 0 if succesful, 1 otherwise
+############################################################
+legacy_transform_is_ancestor() {
+    if [ -n "$1" ] && [ -n "$2" ] &&
+        execute_git "$GITHOOKS_CLONE_DIR" merge-base --is-ancestor \
+            "$1" "$2" >/dev/null 2>&1; then
+        # commit 1 <= commit 2
+        return 0
+    fi
+
+    return 1
+}
+
+############################################################
+# Function to dispatch to all legacy transformations
+#   right after the update.
+#
+# Returns: 0 if succesful, 1 otherwise
+############################################################
+legacy_transform_after_update() {
+    COMMIT_COUNT=$(execute_git "$GITHOOKS_CLONE_DIR" rev-list --count HEAD)
+    GITHOOKS_CLONE_CURRENT_COMMIT=$(execute_git "$GITHOOKS_CLONE_DIR" rev-parse HEAD)
+
+    if [ "$COMMIT_COUNT" != "1" ] && [ -z "$INTERNAL_UPDATED_FROM_COMMIT" ]; then
+        # If the clone dir has been updated (commit count != 1) and
+        # we do not have the update commit yet (meaning we are updating from an old version )
+        # we set it to the last commit where this feature (INTERNAL_UPDATED_FROM_COMMIT)
+        # was not yet available
+        INTERNAL_UPDATED_FROM_COMMIT="ab86d2a529f58744a71e79227e434f19b84589e6"
+    fi
+
+    # Because of changes in PR #125 right after commit ab86d2a5:
+    # - Check if we need transforms for `--global githooks.shared`
+    #   to be split into multiple config values
+    # - Show info that all hook hashes are differently computed now
+    #   and every hooks needs to be trusted again.
+    # - Show info that `githooks.failOnNonExistingHooks` is enabled due to renaming
+    #   internal cloned shared hooks repositories.
+    if legacy_transform_is_ancestor \
+        "$INTERNAL_UPDATED_FROM_COMMIT" \
+        "ab86d2a529f58744a71e79227e434f19b84589e6"; then
+
+        legacy_transform_split_global_shared_entries || LEGACY_TRANSFORM_FAILURES="true"
+
+        echo >&2
+        echo "! Info: Because the hash algorithm changed from" >&2
+        echo "  $(md5sum) to $(git hash-object)," >&2
```

Comment:
This `echo` is going to try evaluating `$(md5sum)` and `$(git hash-object)`, which is not what you wanted here I think.
Maybe it needs escaping?

Reply:
Thx!